### ISSUE TITLE
emcc.py builds wrong path to asm2wasm

### ISSUE THREAD

**Issue by ataulien** [emscripten-core/emscripten#5340]
My project does successfully compile using Emscripten targeting the ususal asm.js. However, it fails when I try to use WASM. This is the output I get right after my data is being packaged:

```
Traceback (most recent call last):
  File "/usr/lib/emscripten/emcc.py", line 2577, in <module>
    run()
  File "/usr/lib/emscripten/emcc.py", line 1792, in run
    wasm_text_target, misc_temp_files, optimizer)
  File "/usr/lib/emscripten/emcc.py", line 2258, in do_binaryen
    subprocess.check_call(cmd)
  File "/usr/lib/python2.7/subprocess.py", line 181, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib/python2.7/subprocess.py", line 168, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/lib/python2.7/subprocess.py", line 390, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1024, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
```

I put a `print`-call right before `subprocess.check_call` to see what the script is actually trying to call, which reported this:
`['/usr/bin/bin/asm2wasm', 'REGoth.temp.asm.js', '--total-memory=838860800', '--emit-potential-traps', '-O2', '--mem-init=REGoth.html.mem', '--mem-base=1024', '--wasm-only', '-g', '-o', 'REGoth.wast', '-S']`

`asm2wasm` does exist in `/usr/bin`, but notice the extra `bin` in the path, which obviously wrong.

Maybe this is just a configuration issue, but since you need to install asm2wasm individually (?), the script may output a better error message if it wasn't found. 

`emcc --version` reports:
```
emcc (Emscripten gcc/clang-like replacement) 1.37.14 ()
```

**Comment by kripken** [comment]
Yeah, sounds like a misconfiguration issue. What does `BINARYEN_ROOT` point to in `~/.emscripten`?

Perhaps we should add a warning when it doesn't exist, saying to check the config file.

**Comment by degenerated1123** [comment]
There it is:
```sh
$ cat .emscripten | grep BINARYEN_ROOT
BINARYEN_ROOT = os.path.expanduser(os.getenv('BINARYEN') or '/usr/bin') # directory
```
and
```sh
$ echo $BINARYEN
/usr
```

I didn't change anything there, at least not manually. I installed Binaryen from the AUR: https://aur.archlinux.org/packages/binaryen/

**Comment by kripken** [comment]
Hmm, if BINARYEN is defined that way, then it should work? It should append `bin` to `/usr`.

Otherwise, the `/usr/bin` detection looks wrong, it should not include `bin` (since we need the binaryen root dir, which contains not just `bin/asm2wasm` but also other files). Fix in #5344.

**Comment by juj** [comment]
@degenerated1123 : out of curiosity, what is the installation process you use to get Emscripten to live at `/usr/lib/emscripten`?
